src/libs/context.js

```diff
@@ -153,6 +160,24 @@ function nameOrFallback () {
   return this.name
 }
 
+/**
+ * determine if an input has a user-defined name
+ */
+function hasGivenName () {
+  if (
+    this.name &&
+    typeof this.name === 'string' &&
+    !['checkbox', 'radio'].includes(this.type) &&
```

Comment:
I'm currently filtering out checkbox and radio types. If you think it makes sense to pass a given name prop down to each checkbox and radio option then we should also be able to do that.

Reply:
What is the thinking around not passing names to inputs and checkboxes?

Reply:
Honestly, I wasn't sure if it made sense to because they would all be identical duplicates. Shouldn't be too hard to output them if it makes sense to.